source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -192,6 +193,14 @@ namespace PlayFab
                     {
                         // Add HTTP headers
                         SetPredefinedHeaders(reqContainer, hRequest);
+                        if(!setPredefinedHeadersFailed)
+                        {
+                            SetErrorInfo(reqContainer, "Error in attempting to add Default Headers with HRESULT: " + std::to_string(GetLastError()));
+                            CompleteRequest(std::move(requestContainer), hRequest, hConnect, hSession);
+                            setPredefinedHeadersFailed = true;
```

Comment:
>setPredefinedHeadersFailed [](start = 28, length = 26)

since you are using std::atomic, I recommend taking advantage of that and using a single `exchange` operation instead of splitting the load and store across two lines of code (which defeats part of the point of using an atomic)

```
SetPredefinedHeaders(..);
bool failed = setPredefinedHeadersFailed.exchange(true); // load the value that SetPredefinedHeaders set and atomically clear it back to true
if (!failed)
{
   SetErrorInfo(...)
   CompleteRequest(...);
}
```

 (note I've left comments elsewhere about how the setPredefinedHeadersFailed is backwards and I suspect you actually want to be exchanging to `false` and comparing the if-check for `true` values, but I've left that alone for this comment to demonstrate what would actually change by using `exchange` instead of a separate load and store)